packages/babel-cli/src/babel/file.js

```diff
@@ -65,7 +76,7 @@ export default async function({ cliOptions, babelOptions }) {
       babelOptions.sourceMaps === "inline" ||
       (!cliOptions.outFile && babelOptions.sourceMaps)
     ) {
-      code += "\n" + convertSourceMap.fromObject(map).toComment();
+      code += "\n" + convertSourceMap.fromObject(map.toJSON()).toComment();
```

Comment:
`toJSON()` will incur extra Object memory allocation here given that `convertSourceMap` can work with `SourceMapGenerator` to generate a base64 comment.

I would suggest we extend the signature of `fromObject` to state that it can work with `SourceMapGenerator`.
https://github.com/babel/babel/blob/fced5cea430cc00e916876b663a8d2a84a5dad1f/lib/third-party-libs.js.flow#L180

Reply:
Alright. I can make that change.